packages/kas/src/parser.ts

```diff
@@ -0,0 +1,387 @@
+import {type Token, type Options, lex} from "./lexer";
+import {
+    Abs,
+    Add,
+    Const,
+    Eq,
+    Float,
+    Func,
+    Int,
+    Log,
+    Mul,
+    Pow,
+    Trig,
+    Var,
+} from "./nodes";
+
+type Expr = any;
+
+export class Parser {
```

Comment:
TODO: Add some comments at the top here about this being a recursive descent parser and how the order that the methods are called determines the precedence of the operators.  Include a list of the current operators and their relative precedences.